src/bootstrap/src/bin/main.rs

```diff
@@ -15,8 +15,20 @@ use bootstrap::{
     human_readable_changes, t,
 };
 use build_helper::ci::CiEnv;
-
+// These are conditionally enabled if the user specifies `BOOTSTRAP_LOG=...`, which builds bootstrap
+// with `--features=logging`.
+#[cfg(feature = "logging")]
+use tracing::*;
+#[cfg(feature = "logging")]
+use tracing_subscriber::EnvFilter;
+#[cfg(feature = "logging")]
+use tracing_subscriber::prelude::*;
```

Comment:
Suggestion: If we unconditionally create modules `crate::logging` and `crate::logging::prelude`, and then conditionally re-export things from those modules, then other modules can just do an unditional `use crate::logging::prelude::*`.

That might reduce the amount of `#[cfg(feature = "logging")]` boilerplate in bootstrap overall.

Reply:
These are needed for the binary, but other places only really need

```rs
#[cfg(feature = "logging")]
use tracing::*;
```

so not sure if worth the extra prelude.

Reply:
That also allows the interesting possibility of declaring dummy `trace!` macros when not in tracing mode, so that individual trace points don't need `[#cfg(..)]`.

Reply:
Wouldn't those `trace!` macros need to actually mimic the `tracing::trace!` macro's multiple forms?

Reply:
Oh, perhaps. I haven't thought through the details, so it might be more complicated than I'm imagining, and thus not worth it.

Reply:
For this initial implementation, let's go with the current least-complicated setup.